bson/_cbsonmodule.c

```diff
@@ -510,8 +536,24 @@ static int write_element_to_buffer(PyObject* self, buffer_t buffer,
     int result;
     if(Py_EnterRecursiveCall(" while encoding an object to BSON "))
         return 0;
+
+    /* Transform types that have a registered converter.
+     * A new reference is created upon transformation. */
+    PyObject* value_type = PyObject_Type(value);
+    PyObject* converter = NULL;
+    int new_reference = 0;
+    if ((converter = PyDict_GetItem(options->type_registry.encoder_map, value_type)) != NULL) {
+        PyObject* args = PyTuple_Pack(1, value);
```

Comment:
I think you can avoid this by using PyObject_CallFunctionObjArgs instead of PyObject_CallObject.

https://docs.python.org/3/c-api/object.html#c.PyObject_CallFunctionObjArgs

Reply:
Nice! I was looking for a method like that but I couldn't find it in the docs.